docs/tuning-apps/networking/cuda.rst

```diff
@@ -410,135 +286,75 @@ example of how to write CUDA-aware MPI applications.
-* MPI_Ssend_init
-* MPI_Win_create
-
-.. FIXME: We need to verify the above list.
 
-/////////////////////////////////////////////////////////////////////////
+.. attention::
 
-Which MPI APIs do NOT work with CUDA-aware?
--------------------------------------------
+    Some blocking collective algorithms are implemented with non-blocking APIs,
+    and therefore lack CUDA-aware support. In such cases, the application can
+    unselect the algorithm or component with corresponding MCA parameters.
+
+One-sided communication APIs currently have limited CUDA-aware support. Only
+``MPI_Get`` and ``MPI_Put`` support GPU buffers.
+
+APIs without CUDA-aware support
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Non-blocking collective communication APIs generally do not have CUDA-aware
+support, plus reductive blocking APIs:
+
+* MPI_Allreduce
+* MPI_Reduce
+* MPI_Reduce_scatter
+* MPI_Reduce_scatter_block
+
+One-sided commnication APIs other than ``MPI_Get`` and ``MPI_Put`` do not fully
+support GPU buffers, including:
 
 * MPI_Accumulate
 * MPI_Compare_and_swap
 * MPI_Fetch_and_op
 * MPI_Get_Accumulate
-* MPI_Iallgather
-* MPI_Iallgatherv
-* MPI_Iallreduce
-* MPI_Ialltoall
-* MPI_Ialltoallv
-* MPI_Ialltoallw
-* MPI_Ibcast
-* MPI_Iexscan
 * MPI_Rget
 * MPI_Rput
 
-.. FIXME: We need to verify the above list.
-
-/////////////////////////////////////////////////////////////////////////
-
-How do I use CUDA-aware UCX for Open MPI?
------------------------------------------
-
-Example of running ``osu_latency`` from the `OSU benchmarks
-<https://mvapich.cse.ohio-state.edu/benchmarks>`_ with CUDA buffers
-using Open MPI and UCX CUDA support:
-
-.. code-block::
-
-   shell$ mpirun -n 2 --mca pml ucx \
-       -x UCX_TLS=rc,sm,cuda_copy,gdr_copy,cuda_ipc ./osu_latency D D
+UCX and UCC
+^^^^^^^^^^^
 
-/////////////////////////////////////////////////////////////////////////
-
-Which MPI APIs work with CUDA-aware UCX?
-----------------------------------------
+UCX and UCC supports CUDA-aware blocking reduction collective APIs:
 
-* MPI_Send
-* MPI_Bsend
-* MPI_Ssend
-* MPI_Rsend
-* MPI_Isend
-* MPI_Ibsend
-* MPI_Issend
-* MPI_Irsend
-* MPI_Send_init
-* MPI_Bsend_init
-* MPI_Ssend_init
-* MPI_Rsend_init
-* MPI_Recv
-* MPI_Irecv
-* MPI_Recv_init
-* MPI_Sendrecv
-* MPI_Bcast
-* MPI_Gather
-* MPI_Gatherv
-* MPI_Allgather
+* MPI_Allreduce
 * MPI_Reduce
 * MPI_Reduce_scatter
 * MPI_Reduce_scatter_block
-* MPI_Allreduce
-* MPI_Scan
-* MPI_Exscan
-* MPI_Allgatherv
-* MPI_Alltoall
-* MPI_Alltoallv
-* MPI_Alltoallw
-* MPI_Scatter
-* MPI_Scatterv
-* MPI_Iallgather
-* MPI_Iallgatherv
-* MPI_Ialltoall
-* MPI_Iialltoallv
-* MPI_Ialltoallw
-* MPI_Ibcast
-* MPI_Iexscan
-
-.. FIXME: We need to verify the above list.  These _SHOULD_ be the same
-   as above.
-
-/////////////////////////////////////////////////////////////////////////
 
-Which MPI APIs do NOT work with CUDA-aware UCX?
------------------------------------------------
+However, the following APIs do not support GPU buffers:
```

Comment:
is this really correct?  at least with recent UCX versions I think some/most of these functions should work.

Reply:
@bosilca @janjust Do you know if UCC currently supports non-blocking collectives with CUDA buffer? I tried single-node and from what I saw it worked.

Also could you help double check the statement about UCX CUDA support status?

Reply:
UCC supports both blocking and non-blocking collectives with cuda buffers

Reply:
UCX supports cuda buffers I think for most, not sure about cuda-ipc but osc ucx does not I think. I have to double check but I'm pretty sure we have issues in accumulate

Reply:
Thanks Tommy! I will update the UCC part first.